### PR TITLE
Fix issue with getRiskFactorsFromPregnancy()

### PR DESCRIPTION
**Pull request by yrimal**
# Description

Fix and unit test for:  medic/cht-core#8144

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* __CHT_CORE_COMPOSE_URL__
* __COUCH_SINGLE_COMPOSE_URL__
* __COUCH_CLUSTER_COMPOSE_URL__
 
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: config/default/test/contact-summary-extras/get_riskfactors_from_pregnancy.spec.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    expect(getRiskFactorsFromPregnancy(report).length).to.equal(0);
+  });
+
+  it(`getRiskFactorsFromPregnancy() returns primary factors only when secondary factors are empty.`, () => {
+    const primary_conditions = 'p1 p2';
+    report.fields.risk_factors.risk_factors_present.primary_condition = primary_conditions;
+    expect(getRiskFactorsFromPregnancy(report)).to.have.members(['p1', 'p2']);
+  });
+
+  it(`getRiskFactorsFromPregnancy() returns secondary factors only when primary factors are empty.`, () => {
```
Comment (dianabarsan) [inline config/default/test/contact-summary-extras/get_riskfactors_from_pregnancy.spec.js:48]: I think the name of this test contradicts the code change that is proposed.

**Review by dianabarsan — APPROVED** [review]
Just a nitpick about a test name.

**Review by jkuester — APPROVED** [review]
:t-rex:   Even included tests!  :heart:

File: config/default/test/contact-summary-extras/get_riskfactors_from_pregnancy.spec.js (hunk 1 above)
Comment (yrimal) [reply config/default/test/contact-summary-extras/get_riskfactors_from_pregnancy.spec.js:48]: I added tests apart from the issue fix as well and wanted to ensure that the function works as intended, returning all risk factors based on all combination.

Comment (dianabarsan) [reply config/default/test/contact-summary-extras/get_riskfactors_from_pregnancy.spec.js:48]: I don't disagree with the test itself, but the title says `returns secondary factors only when primary factors are empty`. 
Is this still true?

Comment (yrimal) [reply config/default/test/contact-summary-extras/get_riskfactors_from_pregnancy.spec.js:48]: Yes. Initially `secondary` tests were being returned only when `primary factors` were empty. This [test](https://github.com/medic/cht-core/pull/8145/files#diff-3fea9cc2cd330e283fada5f9e7dbd6c9f1f15a4ae07c1198f9afbaf45da4ae34R54-R57) ensures that secondary factors are also returned even if primary risk factors exist, actual fix for issue. 

This test and [this](https://github.com/medic/cht-core/pull/8145/files#diff-3fea9cc2cd330e283fada5f9e7dbd6c9f1f15a4ae07c1198f9afbaf45da4ae34R42-R46) test that in one's absence other risk factors are returned.

**Pull request merged**